index.bs

```diff
@@ -692,44 +692,75 @@ When this method is invoked, the user agent MUST execute the following algorithm
 
 1. For each |authenticator| currently available on this platform, perform the following steps:
 
-    <span class="note">Note: the procedure to determine the [=authenticators=] present on a platform is 
-        platform-specific.</span>
-
-    1. Let |credentialList| be a new [=ordered set=].
+    1. Let |credentialDescriptorList| be a new [=ordered set=] of {{ScopedCredentialDescriptor}}.
 
     1. If <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code>:
         <dl class="switch">
             : [=list/is not empty=]
-            :: [=set/For each=] credential |C| in <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code>:
-                1. If there is any credential |X| present on this |authenticator| that:
+            :: [=set/For each=] {{ScopedCredentialDescriptor}} |C| in
+                <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code>:
+                1. For each credential |X| present on this |authenticator| that:
                     - is mapped to |rpId|, and,
                     - matches with <code>|C|.{{ScopedCredentialDescriptor/id}}</code> and 
                         <code>|C|.{{ScopedCredentialDescriptor/type}}</code>,
 
-                    then [=set/append=] |X| to |credentialList|.
+                    then:
+                        1. let |Y| be a new {{ScopedCredentialDescriptor}}.
+
+                        1. Set |Y|'s members according to the corresponding properties of |X|.
 
-                <span class="note">Note: the procedures to determine which credentials present on |authenticator| map to 
-                    |rpId|, and to match credentials present on |authenticator| with values in
-                    {{ScopedCredentialRequestOptions/allowList}}, are platform-specific.</span>
+                        1. [=set/append=] |Y| to |credentialDescriptorList|.
 
             : [=list/is empty=]
-            :: If there is any credential |X| present on this |authenticator| that is mapped to |rpId| then [=set/append=] |X| 
-                to |credentialList|.
+            :: For each credential |X| present on this |authenticator| that is mapped to |rpId| then:
+
+                1. let |Y| be a new {{ScopedCredentialDescriptor}}.
+
+                1. Set |Y|'s members according to the corresponding properties of |X|.
```

Comment:
This looks good, except it's unclear about |Y|.transports, since |X| doesn't have any. I suspect you want |Y|.transports to be empty? Then I'd spell this:
> [=set/Append=]  «[ "id" → |X|'s credential ID, "type" → |X|'s type ]» to |credentialDescriptorList|.